src/test/java/rx/internal/operators/OperatorThrottleFirstTest.java

```diff
@@ -172,4 +174,44 @@ public void timed() {
         ts.assertNoErrors();
         ts.assertCompleted();
     }
+
+    @Test
+    public void testThrottleWithoutAdvancingTimeOfTestScheduler() {
+        @SuppressWarnings("unchecked")
+        Observer<Integer> observer = mock(Observer.class);
+        TestScheduler s = new TestScheduler();
+        PublishSubject<Integer> o = PublishSubject.create();
+        o.throttleFirst(500, TimeUnit.MILLISECONDS, s).subscribe(observer);
+
+        // send events without calling advanceTimeBy/To
+        o.onNext(1); // deliver
+        o.onNext(2); // skip
+        o.onNext(3); // skip
+        o.onCompleted();
+
+        verify(observer).onNext(1);
+        verify(observer).onCompleted();
+        verifyNoMoreInteractions(observer);
+    }
+
+    @Test
+    public void testThrottleWithTestSchedulerTimeOfZero() {
```

Comment:
Ok, that's reasonable

On Sun, 21 Aug 2016, 22:04 Anton Rutkevich, <notifications@github.com>
wrote:

> In src/test/java/rx/internal/operators/OperatorThrottleFirstTest.java
> <https://github.com/ReactiveX/RxJava/pull/4397#discussion_r75601379>:
>
> > +        PublishSubject<Integer> o = PublishSubject.create();
> > +        o.throttleFirst(500, TimeUnit.MILLISECONDS, s).subscribe(observer);
> > +
> > +        // send events without calling advanceTimeBy/To
> > +        o.onNext(1); // deliver
> > +        o.onNext(2); // skip
> > +        o.onNext(3); // skip
> > +        o.onCompleted();
> > +
> > +        verify(observer).onNext(1);
> > +        verify(observer).onCompleted();
> > +        verifyNoMoreInteractions(observer);
> > +    }
> > +
> > +    @Test
> > +    public void testThrottleWithTestSchedulerTimeOfZero() {
>
> The intention was not to rely implicitly on the fact that TestScheduler's
> advanceTimeBy(0) is a no-op.
> I don't think it's absolutely necessary, though.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/4397/files/36cf057a7eaea1e013eb320ccbb644841fa75f77#r75601379>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA7B3KW6ptfbQe-Y1WMl8cfndGK8ZPlMks5qiKEsgaJpZM4JpUFZ>
> .
>